packages/api/src/services/save_page.ts

```diff
@@ -139,14 +143,37 @@ export const savePage = async (
       }
     }
   } else {
-    if (!(await createPage(articleToSave, ctx))) {
+    pageId = await createPage(articleToSave, ctx)
+    if (!pageId) {
       return {
         errorCodes: [SaveErrorCode.Unknown],
         message: 'Failed to create new page',
       }
     }
   }
 
+  if (pageId && parseResult.highlightData) {
+    const highlight = {
+      updatedAt: new Date(),
+      createdAt: new Date(),
+      userId: ctx.uid,
+      elasticPageId: pageId,
+      ...parseResult.highlightData,
+    }
+
+    if (
+      !(await addHighlightToPage(pageId, highlight, {
```

Comment:
Not sure if it will help to speed up a bit if we save the highlight with the page together in one call

Reply:
Yeah maybe for a future optimization.